ports/psoc6/machine_rtc.c

```diff
@@ -175,16 +172,22 @@ static inline uint64_t rtc_get_current_time_in_sec() {
 
 }
 
-void rtc_irq_handler1(void *callback, cyhal_rtc_event_t event) {
-    mp_call_function_1((mp_obj_t)callback, mp_obj_new_int(event));
-}
+static void rtc_irq_handler(void *self, cyhal_rtc_event_t event);
 
-void rtc_irq_handler2(void *callback, cyhal_rtc_event_t event) {
-    machine_rtc_obj_t *self = (machine_rtc_obj_t *)callback;
-    cyhal_rtc_register_callback(&psoc6_rtc, (cyhal_rtc_event_callback_t)rtc_irq_handler2, self);
-    cyhal_rtc_enable_event(&psoc6_rtc, CYHAL_RTC_ALARM, 3, true);
+static void rtc_alarm_setup(machine_rtc_obj_t *self) {
     cyhal_rtc_set_alarm_by_seconds(&psoc6_rtc, mp_obj_get_int(self->alarm_period_s));
-    mp_call_function_1((mp_obj_t)self->callback, mp_obj_new_int(event));
+    cyhal_rtc_register_callback(&psoc6_rtc, (cyhal_rtc_event_callback_t)rtc_irq_handler, self);
+    cyhal_rtc_enable_event(&psoc6_rtc, CYHAL_RTC_ALARM, 3, true);
+}
+
+void rtc_irq_handler(void *self_in, cyhal_rtc_event_t event) {
+    machine_rtc_obj_t *self = (machine_rtc_obj_t *)self_in;
+    if (self->callback != NULL) {
```

Comment:
But where is this initialized as NULL? Should be done in cases where there is no alarm or there is alarm but no callback associated.